### PR TITLE
fix: empty the success hosts and correctly set missing_reqs

### PR DESCRIPTION
**Pull request by Tiboris**
Since I have coded a regression when trying to re-provision
mrack is not setting variables for another loop correctly
error_hosts is updated correctly but when keeping success_hosts
not changed to empty list it creates a illusion of hosts
which were deleted because of re-provisioning strategy
that they are still up and running.

### CODE REVIEW

File: src/mrack/providers/provider.py [hunk 1]
```diff
@@ -413,6 +409,12 @@ async def provision_hosts(self, reqs):
 
         return success_hosts
 
+    def _get_missing_reqs(self, reqs, error_hosts):
+        """Rerurn missing requirements based on reqs and error hosts."""
```
Comment (pvoborni) [inline src/mrack/providers/provider.py:413]: Typo in the description.

**Review by pvoborni — CHANGES_REQUESTED** [review]
The code looks good to me. But it is hard to tell whether it indeed works.  This kind of scenario is quite difficult to reproduce in adhoc testing. A test with good usage of OpenStack calls mocking which would simulate the scenario would be great to have.

Also the commit/MR description doesn't really describe the scenario which is not working.

File: src/mrack/providers/provider.py (hunk 1 above)
Comment (Tiboris) [reply src/mrack/providers/provider.py]: fixed and updated commit message.

**Review by pvoborni — APPROVED** [review]
I'm approving this as the commit has better message and I saw that the PR really helps in internal real testing. 

That said we still need to deliver the tests soon, but not in this PR.

**Pull request merged**